Update README.md (#52)
floodnet-nyc/flood-sensor: firmware/README.md

### Arduino IDE
For simplicity, Arduino IDE has been used in the documentation but other IDEs can also be used. Arduino IDE, can be downloaded from [Arduino's website](https://www.arduino.cc/en/software). The Arduino IDE is used for uploading the firmware to the microcontroller.

Add HelTec Boards to the IDE, there are two ways:
 - Using Arduino's Boards Manager.

>      - Open Arduino IDE, and goto File->Preferences->Settings

Reviewer: ```suggestion
     - Open Arduino IDE, and goto `File->Preferences->Settings`
```

      <br> 
      <img src="img/IDE preferences.png" width="440" height="300"> 
     
     - Enter the [URL](https://github.com/HelTecAutomation/CubeCell-Arduino/releases/download/V1.5.0/package_CubeCell_index.json) to board manager URLs. Then hit 		 ok.

>      - After that goto Tools->Board:->Boards Manager, search CubeCell in the new pop-up dialog, select the latest releases and click install. 

Reviewer: ```suggestion
     - After that goto `Tools->Board->Boards Manager`, and search CubeCell. In the new pop-up dialog select the latest releases and click install. 
```

      <br>
      <img src="img/IDE.png" width="900" height="200">
 - Using Github.
     - Follow the instructions mentioned [here.](https://github.com/HelTecAutomation/CubeCell-Arduino/blob/master/InstallGuide/windows.md)
     </a>

> Download the Flood Sensor Library from [here](https://github.com/floodnet-nyc/flood-sensor). Following are the steps to install this library:

Reviewer: remove lines 42-44

Reviewer: ```
Download the Flood Sensor Library from [here](https://github.com/floodnet-nyc/flood-sensor). Following are the steps to install this library:
1......
2....
```

1. Find Arduino libraries folder
2. Copy the `firmware` folder into the Arduino>libraries

### Installing CubeCell Library
This repo is built to run on CubeCell's example library. To install the CubeCell's library, follow the instructions on their repo [here](https://github.com/HelTecAutomation/CubeCell-Arduino/tree/master/InstallGuide)

### Registering the device to The Things Network
 Setting up The Things Network

>   - Make an account on The Things Network at [here](https://www.thethingsnetwork.org/)

Reviewer: Add a step here to demonstrate logging into the TTN console and provide a link to the console `https://nam1.cloud.thethings.network/console/`

  - If the target application at which your device will be registered is not created yet, follow this step.
      - Goto Application-> Create application.
      - Enter Application ID(should be unique) and Application Name.
 
      
### Creating an end-device
 Registering End Device.
   - Goto the application then click on ‘Register end device’. 
   - Select the end device in the LoRaWAN Device Repository. Refer the following image.
     <br> 
     <img src="img/End Device.png" width="550" height="300"> 

>    - In JoinEUI an ID has to be entered which can be anything (for example, FA CD 67 98 HY 7J 54 00). Once entered, additional DevEUI and AppKey have to be

Reviewer: ```suggestion
   - For the `JoinEUI` field, random ID can be used except all zeroes (for example, `FA CD 67 98 HY 7J 54 00` can be used but not `00 00 00 00 00 00 00 00`). Next, the fields `DevEUI` and `AppKey` needs to be 
```

     entered which can be generated by clicking on generate. And then click on Register end Device. 

## Usage
 To use this library, user can simply open firmware.ino from the firmware folder.
### Installing the device secrets
 - Connecting CubeCell to the Arduino IDE.

>     - Open Arduino IDE and goto Tools>Board>CubeCell>HTTC(AB02). And set the configurations according to the image below.

Reviewer: ```suggestion
    - Open Arduino IDE and goto `Tools>Board>CubeCell>HTTC(AB02)`. And set the configurations according to the image below.
```

      <br> 
      <img src="img/board specs1.png" width="500" height="300"> 
    - Connect the HelTec CubeCell board to the computer via a high quality Micro USB cable. CP2102 driver is required, most of the time it will install
      automatically and if didn't install automatically, please refer to this [document.](https://docs.heltec.cn/general/establish_serial_connection.html).

>     - Select the port at which CubeCell is connected to computer by going to Tools>Port.

Reviewer: ```suggestion
    - Select the port at which CubeCell is connected to computer by going to `Tools>Port`.
```

 - EEPROM should be cleared every time when changes are made to device secrets. Clear the EEPROM which you have to upload the following example.

>       Goto File->Examples->EEPROM->eeprom_clear and upload the code by clicking on the right arrow symbol.

Reviewer: ```suggestion
      Goto `File->Examples->EEPROM->eeprom_clear` and upload the code by clicking on the right arrow symbol.
```

 - Adding device secerts in the firmware.
    - Goto File->Examples->firmware. This will open all the files reuired by the sensor to be working.
    - Open device_secrets.h file from the tab.

>     - Enter the DevEUI, JoinEUI and AppKey in TTN_DEVEUI, TTN_APPEUI AND TTN_APPKEY respectively. Refer the image. 

Reviewer: ```suggestion
    - Enter the `DevEUI`, `JoinEUI` and `AppKey` in `TTN_DEVEUI`, `TTN_APPEUI`, and `TTN_APPKEY` respectively like shown in the image below 
```
